docs/04.guides/04.cookbooks/29.Threads-Scope/page.md

```diff
@@ -0,0 +1,272 @@
+---
+title: Threads
+id: thread_usage
+---
+## Thread Scope ##
+
+This document explains how to use the threads. Threads mainly used for execute the code parallel.
+
+### Example 1 ###
+
+This below example shows waiting normal execution waiting for all code to resolve. Here it waits for 1000 mili seconds after the it complete the execution.
+
+```lucee
+<cfscript>
+function napASecond() localmode=true {
+	sleep(1000);
+}
+start=getTickCount();
+napASecond();
+dump("done in #getTickCount()-start#ms");
+</cfscript>
+```
+
+This below example shows thread execution. Code with in the thread tag run parallel to execution. Here execution doesn't wait for sleep.
+
+```lucee
+<cfscript>
+function napASecond() localmode=true {
+	thread {
+		sleep(1000);
+	}
+}
+start=getTickCount();
+napASecond();
+dump("done in #getTickCount()-start#ms");
+</cfscript>
+```
+Threads were independent form current running thread on lucee. Lucee doesn't wait for end of thread.
+
+Threads were mainly used to retrieve the data form DB, cfhttp, webservice it doesn't care about how much time it takes to execute.
```

Comment:
***ToChange***
***This comment is for lines 38 and 40***
Threads run independently of other threads or code in Lucee. Lucee does not the thread first and then the following code.

Threads are mainly used to retrieve data from the database, cfhttp, or webservice. Threads are used when it does not matter how much time it takes to execute.